sdk/core/Azure.Core/src/Shared/LightweightPkcs8Decoder.cs

```diff
@@ -31,6 +32,160 @@ internal class LightweightPkcs8Decoder
             0x05, 0x00,
         };
 
+        internal static byte[] ReadBitString(byte[] data, ref int offset)
```

Comment:
PKCS#8 says that the contents are BER encoded, which technically means that the constructed encodings of BIT STRING and OCTET STRING are permissible.  Neither of these routines support the constructed encodings.

I'm not recommending you try to support them, and they'll probably never come up in practice, but it is one loss of fidelity you have by not using AsnReader here.

Reply:
Good to know. We'll keep an eye on customer-reported issues and fix this if we need to. Hopefully before then we can switch to the `AsnReader`.